Tests/HAPTests/EndpointTests.swift

```diff
@@ -202,9 +204,407 @@ class EndpointTests: XCTestCase {
+            let response = application(MockConnection(), MockRequest(method: "PUT", path: "/characteristics", body: body))
+            XCTAssertEqual(response.status, .unprocessableEntity)
+        }
+        
+        do {
+            let jsonObject: [String: [[String: Any]]] = [
+                "characteristics": [
+                    [
+                        "aid": 1,
+                        "iid": 33,
+                        "value": "value"
+                    ]
+                ]
+            ]
+            let body = try! JSONSerialization.data(withJSONObject: jsonObject, options: [])
+            let response = application(MockConnection(), MockRequest(method: "PUT", path: "/characteristics", body: body))
+            XCTAssertEqual(response.status, .unprocessableEntity)
+        }
+
+
+        
     }
+    
+    func testGetBadCharacteristics() {
+        let lightsensor = Accessory.LightSensor(info: .init(name: "LightSensor"))
+        let thermostat = Accessory.Thermostat(info: .init(name: "Thermostat"))
+        let lamp = Accessory.Lightbulb(info: .init(name: "Night stand left"))
+        
+        lightsensor.lightSensor.currentLight.value = 234
+        thermostat.thermostat.currentTemperature.value = 123
+        lamp.lightbulb.brightness.value = 53
+        let device = Device(name: "Test", pin: "123-44-321", storage: MemoryStorage(), accessories: [lightsensor,thermostat,lamp])
+        let application = characteristics(device: device)
+        
+        // First a good one
+        
+        do {
+            let req = "\(lamp.aid).\(lamp.lightbulb.brightness.iid),\(lightsensor.aid).\(lightsensor.lightSensor.currentLight.iid),\(thermostat.aid).\(thermostat.thermostat.currentTemperature.iid)"
+            let response = application(MockConnection(), MockRequest.get(path: "/characteristics?id=\(req)"))
+            
+            XCTAssertEqual(response.status, .ok)
+            
+            guard let jsonObject = (try? JSONSerialization.jsonObject(with: response.body!, options: [])) as? [String: [[String: Any]]] else {
+                return XCTFail("Could not decode")
+            }
+            guard let characteristics = jsonObject["characteristics"] else {
+                return XCTFail("No characteristics")
+            }
+            
+            guard let light = characteristics.first(where: { $0["aid"] as? Int == lightsensor.aid }) else {
+                return XCTFail("Could not get light aid")
+            }
+            
+            guard let therm = characteristics.first(where: { $0["aid"] as? Int == thermostat.aid }) else {
+                return XCTFail("Could not get therm aid")
+            }
+            
+            guard let lampa = characteristics.first(where: { $0["aid"] as? Int == lamp.aid }) else {
+                return XCTFail("Could not get lampa aid")
+            }
+            
+            guard let lightVal = light["value"] as? Double else {
```

Comment:
On Linux, decoding from JSON is a bit more involved and that's why the test is probably failing here. When there's no decimal separator in the serialised value, SwiftFoundation decodes it as an Int, instead of a Double. See also [SR-4599](https://bugs.swift.org/browse/SR-4599). I've resolved it in HAP by also supporting initialisation of Int in 25a7afd9e8948ff47db285dfe1b3d5249f563f29, which is used at [Characteristic.swift#L71](https://github.com/Bouke/HAP/blob/ccd4e263adce8993f39394bed7b3a026bc3c133c/Sources/HAP/Base/Characteristic.swift#L71).

Reply:
Seeing that `Double(value: Any)` is public, you should be able to rewrite this line (and the others below) as:

    guard let lightVal = Double(value: light["value"]) else {

Reply:
Thanks! I had difficulties to understand the reason for the error, uploaded 95c58ef43d1edf6ec17f6f770bed2ee9d8ce39cc that hopefully will resolve the problem. I have currently no Linux installation so I can't test the code.